### ISSUE TITLE
some macos binary improvement requests to ease packaging and TCC/Startup items admin burden

### ISSUE THREAD

**Issue by ghals13** [Velocidex/velociraptor#3272]
Hello, 

For the macos versions of VRR - could you please consider making some changes to make packaging and maintaining the MDM PPC profile for Full disk access and Startup items a bit easier. 

1. Please consider using a fixed identifier in your code signing profile. It greatly simplifies updating TCC (and doesn't force ommiting the identifier )

```
identifier "velociraptor-v0.7.1-1-darwin-arm64" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = UL6CGN7MAL
```

vs (for example) 

```
identifier "com.velocidex.velociraptor" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = UL6CGN7MAL
```

2. Please consider asking R7 to use a dedicated cert to Sign VRR. As far as I know it's not expensive but I can see there's at least 20 other thing other than VRR signed with it like Electron, some other R7 agent things - not entirely sure they can vouch for all of them being defensive :) .  https://www.virustotal.com/gui/search/signature%253A%2520UL6CGN7MAL/files 

3. Please consider fixing the error below we get when we try to notarize the signed macos package with the VRR bins inside it gets Rejected by apple as invalid - seems like a hardening build/ compilation time flag? 

```
  "severity": "error",
      "code": null,
      "path": "vrr_pkg_signed.pkg/vrr_pkg_signed.pkg Contents/Payload/opt/vrr_installer/velociraptor-v0.7.1-1-darwin-arm64",
      "message": "The executable does not have the hardened runtime enabled.",
      "docUrl": "https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/resolving_common_notarization_issues#3087724",
```
	  

4. Please also consider including  the 'fat' (universal?) mac binary (combined arm64+i386 aka amd64) in releases. However that might be a little bit debatable In terms of slightly larger disk footprint/initial package size to pull, and I'm not entirely sure how much it factors into memory footprint  (I mean I hope they load just the platform that's needed into memory). To each their own some people will use it others make 2 packages :)

**Comment by scudette** [comment]
[velociraptor-v0.7.1-1-darwin-amd64.gz](https://github.com/Velocidex/velociraptor/files/14205226/velociraptor-v0.7.1-1-darwin-amd64.gz)

Thanks for the feedback. I build this file with the codesign options `-i \"com.velocidex.velociraptor\" --options runtime`

Can you please check if it helps?

**Comment by ghals13** [comment]
yep that looks better thanks Mike 

```
Executable=velociraptor-v0.7.1-1-darwin-amd64 2
designated => identifier "com.velocidex.velociraptor" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = UL6CGN7MAL
```

**Comment by scudette** [comment]
This should be fixed with 0.7.1-2 which is out now
